goldenmaster/Plugins/TbEnum/Source/TbEnum/Public/Generated/OLink/TbEnumEnumInterfaceOLinkAdapter.h

```diff
@@ -0,0 +1,84 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#pragma once
+
+#include "TbEnumEnumInterfaceInterface.h"
+#include "Subsystems/GameInstanceSubsystem.h"
+#include "UnrealOLinkHost.h"
+#include "TbEnumEnumInterfaceOLinkAdapter.generated.h"
+
+/// @brief handles the adaption between the service implementation and the OLink protocol
+/// takes an object of the type ITbEnumEnumInterfaceInterface
+/// and holds the corresponding TbEnumEnumInterfaceOLinkSource OLink source object
+UCLASS(BlueprintType)
+class TBENUM_API UTbEnumEnumInterfaceOLinkAdapter : public UGameInstanceSubsystem
+{
+	GENERATED_BODY()
+public:
+	explicit UTbEnumEnumInterfaceOLinkAdapter();
+	virtual ~UTbEnumEnumInterfaceOLinkAdapter() = default;
+
+	// subsystem
+	void Initialize(FSubsystemCollectionBase& Collection) override;
+	void Deinitialize() override;
+
+	UFUNCTION(BlueprintCallable, Category = "ApiGear|TbEnum|EnumInterface")
+	void setBackendService(TScriptInterface<ITbEnumEnumInterfaceInterface> InService);
+
+	UFUNCTION(BlueprintCallable, Category = "ApiGear|TbEnum|EnumInterface")
+	void setOLinkHost(TSoftObjectPtr<UUnrealOLinkHost> InHost);
+
+private:
+	// signals
+	UFUNCTION(Category = "ApiGear|TbEnum|EnumInterface", BlueprintInternalUseOnly)
+	void OnSig0(ETbEnumEnum0 Param0);
+
+	UFUNCTION(Category = "ApiGear|TbEnum|EnumInterface", BlueprintInternalUseOnly)
+	void OnSig1(ETbEnumEnum1 Param1);
+
+	UFUNCTION(Category = "ApiGear|TbEnum|EnumInterface", BlueprintInternalUseOnly)
+	void OnSig2(ETbEnumEnum2 Param2);
+
+	UFUNCTION(Category = "ApiGear|TbEnum|EnumInterface", BlueprintInternalUseOnly)
+	void OnSig3(ETbEnumEnum3 Param3);
+
+	UFUNCTION(Category = "ApiGear|TbEnum|EnumInterface", BlueprintInternalUseOnly)
+	void OnProp0Changed(ETbEnumEnum0 Prop0);
+
+	UFUNCTION(Category = "ApiGear|TbEnum|EnumInterface", BlueprintInternalUseOnly)
+	void OnProp1Changed(ETbEnumEnum1 Prop1);
+
+	UFUNCTION(Category = "ApiGear|TbEnum|EnumInterface", BlueprintInternalUseOnly)
+	void OnProp2Changed(ETbEnumEnum2 Prop2);
+
+	UFUNCTION(Category = "ApiGear|TbEnum|EnumInterface", BlueprintInternalUseOnly)
+	void OnProp3Changed(ETbEnumEnum3 Prop3);
+
+	/** Holds the service backend, can be exchanged with different implementation during runtime */
+	UPROPERTY(VisibleAnywhere, Category = "ApiGear|TbEnum|EnumInterface")
+	TScriptInterface<ITbEnumEnumInterfaceInterface> BackendService;
+
+	/** Holds the registry */
+	TSharedPtr<ApiGear::ObjectLink::RemoteRegistry> Registry;
+
+	/**
+	holds the olink source interface implementation
+	must be std::shared_ptr since it is expected by the olink protocol implementation
+	therefore we need to hold it as member and cannot easily inherit from the source interface
```

Comment:
MINOR, just my opinion, whole comment is good, the last line is not necessary

Reply:
done